src/store/auth/sagas.ts

```diff
@@ -102,15 +103,59 @@ function * loginUserWorker () {
     const user = profile.user;
     const userId = user.id;
     const userName = user.name.split(" ");
-    const currentOrg = profile.current_org;
+
+    let currentOrg = {
+      id: "",
+      name: "",
+      role: {
+        id: "",
+        name: ROLES.baseUser.value,
+      },
+    };
+
+    const currentUser = getFromStorage(LOCAL_STORAGE_KEYS.CURRENT_USER);
+    
+    if (!currentUser) {
+      setInStorage(LOCAL_STORAGE_KEYS.CURRENT_USER, userId);
+    } else if (currentUser !== userId) {
+      removeFromStorage(LOCAL_STORAGE_KEYS.STORED_ORG);
+
+      setInStorage(LOCAL_STORAGE_KEYS.CURRENT_USER, userId);
+    }
+
+    const storedOrg = getFromStorage(LOCAL_STORAGE_KEYS.STORED_ORG);
+
+    if (storedOrg) {
+      currentOrg = {
+        id: storedOrg.id,
+        name: storedOrg.name,
+        role: {
+          id: storedOrg.role.id,
+          name: storedOrg.role.name,
+        },
+      };
+    } else if (profile.organizations.length) {
+      const org = profile.organizations[0];
+
+      currentOrg = {
+        id: org.id,
+        name: org.name,
+        role: {
+          id: org.role.id,
+          name: org.role.name,
+        },
+      };
+    }
+
```

Comment:
I suggest this change to avoid duplication.

```suggestion
    let org = storeOrg
   if (!org && profile.organizations.length) {
      org = profile.organizations[0];
   }
   const currentOrg = {
      id: org?.id || "",
      name: org?.name || "",
      role: {
        id: org?.role?.id || "",
        name: org?.role?.name || ROLES.baseUser.value,
      },
   };
```